docs/workspaces.md

```diff
@@ -0,0 +1,336 @@
+```
+
+Here a `ConfigMap` named `my-configmap` is used for a `Pipeline`'s `Workspace`
+named `myworkspace`:
+
+```yaml
+workspaces:
+- name: myworkspace
+  configmap:
+    name: my-configmap
+```
+
+In this example a `Secret` named `my-secret` is used for a `Pipeline`'s `Workspace`
+named `myworkspace`:
+
+```yaml
+workspaces:
+- name: myworkspace
+  secret:
+    secretName: my-secret
+```
+
+### VolumeSources for Workspaces
+
+Only one type of `VolumeSource` can be used per Workspace entry and the configuration
+options differ for each type. These are the field names for the supported types as well
+as links to the Kubernetes documentation for configuration options of each:
+
+- `emptyDir` - a temporary directory that only lives as long as the TaskRun.
+  - **Note: `EmptyDir` volumes are not suitable for sharing data between `Task`s in a `Pipeline`.**
+  - [Kubernetes docs for `emptyDir` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)
+- `persistentVolumeClaim` - a reference to a `PersistentVolumeClaim`
+  - `PersistentVolumeClaim`s are a good choice for sharing data between `Task`s in a `Pipeline`.
+  - [Kubernetes docs for `persistentVolumeClaim` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#persistentvolumeclaim)
+- `configMap` - a reference to a `ConfigMap`
+  - Using a `ConfigMap` as a `Workspace` has the following caveats:
+    1. `ConfigMap` volume sources are always mounted as read-only. `Step`s
+    cannot write content to them and may error out if a write is attempted.
+    2. The `ConfigMap` you want to use as a `Workspace` must exist prior to
+    the `TaskRun` being submitted.
+    3. `ConfigMap`s have a [size limit of 1MB](https://github.com/kubernetes/kubernetes/blob/f16bfb069a22241a5501f6fe530f5d4e2a82cf0e/pkg/apis/core/validation/validation.go#L5042).
+  - [Kubernetes docs for `configMap` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#configmap)
+- `secret` - a reference to a `Secret`
+  - Using a `Secret` as a `Workspace` has the following caveats:
+    1. `Secret` volume sources are always mounted as read-only. `Step`s
+    cannot write content to them and may error out if a write is attempted.
+    2. The `Secret` you want to use as a `Workspace` must exist prior to
+    the `TaskRun` being submitted.
+    3. `Secret`s have a [size limit of 1MB](https://github.com/kubernetes/kubernetes/blob/f16bfb069a22241a5501f6fe530f5d4e2a82cf0e/pkg/apis/core/validation/validation.go#L5042).
+  - [Kubernetes docs for `secret` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#secret)
+
+> If you need support for a `VolumeSource` not listed here
+> [please open an issue](https://github.com/tektoncd/pipeline/issues) or feel free to
+> [contribute a PR](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md).
+
+## Examples
+
+- [TaskRun example](../examples/v1beta1/taskruns/workspace.yaml)
+- [PipelineRun example](../examples/v1beta1/pipelineruns/workspace.yaml)
```

Comment:
the link results in `404`, it should be  [PipelineRun example](../examples/v1beta1/pipelineruns/workspaces.yaml)

Reply:
PipelineRuns example is named `../examples/v1beta1/pipelineruns/workspaces.yaml` instead of `../examples/v1beta1/pipelineruns/workspace.yaml`

Reply:
For the sake of expediency, I'll do an editorial pass on this new doc once you merge the PR. LGTM for now.